front/lib/api/files/utils.ts

```diff
@@ -116,7 +117,21 @@ export async function maybeUpsertFileAttachment(
             conversationId: conversation.sId,
           });
 
-          return processAndUpsertToDataSource(auth, { file: fileResource });
+          const r = await processAndUpsertToDataSource(auth, {
+            file: fileResource,
+          });
+          if (r.isErr()) {
+            // For now, silently log the error
```

Comment:
Why? we will likely forget about it and regret it no?

Reply:
Intent: if we fail, we'll include the whole file (current behavior) as a fallback.
WDYT ?

Reply:
But do we fail?

Unrelated and meta but the comment at the top of this function:
```
// When we send the attachments at the conversation creation, we are missing the useCaseMetadata
// Therefore, we couldn't upsert them to the conversation datasource.
// We now update the useCaseMetadata and upsert them to the conversation datasource.
```
Is quite confusing as it does not describe the current state of the code but rather a previous state of affair that we don't really need to care about no?

Reply:
> But do we fail?

We could have the upsert or the tokenize or summarization failing intermittently for whatever reason.

It's more a matter of product decision imho, with JIT enabled, do we want to fallback to a degraded mode with the full file included to be resilient ?

> Is quite confusing as it does not describe the current state of the code but rather a previous state of affair that we don't really need to care about no?

???

It is the current state of affair, we will are not able to send the conversationId at creation (first message + attachment problem), this is where we handle it.

Did I misunderstood your question ?